pygmt/base_plotting.py

```diff
@@ -1922,3 +1922,240 @@ def update_pointers(data_pointers):
+
+
+        Other Parameters
+        ----------------
+        vector : bool or str
+            Modify vector parameters. For vector heads, append vector head size
+            [Default is 9p]. For specifying additional attributes, see
+            :gmt-docs:`supplements/geodesy/velo.html#vector-attributes`.
+
+        {B}
+
+        {CPT}
+
+        rescale : str
+            can be used to rescale the uncertainties of velocities
+            (``scaling='e'`` and ``scaling='r'``) and rotations
+            (``scaling='w'``). Can be combined with the confidence variable.
+
+        uncertainty_color : str
+            Sets the color or shade used for filling uncertainty wedges
+            (``scaling='w'``) or velocity error ellipses (``scaling='e'`` or
+            ``scaling='r'``). [If *uncertainty_color* is not specified, the
+            uncertainty regions will be transparent]. More details on
+            :gmt-docs:`cookbook/features.html#gfill-attrib`.
+
+        facecolor : str
+            Select color or pattern for filling of symbols or polygons
+            [Default is no fill]. More details on
+            :gmt-docs:cookbook/features.html#gfill-attrib`.
+
+        line: str
+            Draw lines. Ellipses and fault planes will have their outlines
+            drawn using current pen (see *pen*).
+
+        no_clip: str
+            Do NOT skip symbols that fall outside the frame boundary specified
+            by *region*. [Default plots symbols inside frame only].
+
+        pen : str
+            Set pen attributes for velocity arrows, ellipse circumference and
+            fault plane edges. [Defaults: width = default, color = black,
+            style = solid].
+
+        {U}
+        {V}
+        {XY}
+        {p}
+        {t}
+        """
+        kwargs = self._preprocess(**kwargs)
+
+        kind = data_kind(data)
+
+        extra_arrays = []
+        if "G" in kwargs and not isinstance(kwargs["G"], str):
+            if kind != "vectors":
+                raise GMTInvalidInput(
+                    "Can't use arrays for color if data is matrix or file."
+                )
+            extra_arrays.append(kwargs["G"])
+            del kwargs["G"]
+
+        if scaling is not None:
+            if not isinstance(scaling,str):
+                raise GMTInvalidInput(
+                    "scaling has to be a string."
+                )
+            extra_arrays.append(scaling)
+
+        with Session() as lib:
+            # Choose how data will be passed in to the module
+            if kind == "file":
+                file_context = dummy_context(data)
+            elif kind == "matrix":
+                if pd.api.types.is_numeric_dtype(data):
+                    file_context = lib.virtualfile_from_matrix(data)
+                else:
+                    file_context = lib.virtualfile_from_vectors(
+                        *[data[column] for column in data]
+                    )
```

Comment:
> I haven't read this PR carefully. Do NumPy arrays work before and suddenly don't work after PR #684?

Actually, NumPy arrays will work if there **is not** a text column:

```python
if pd.api.types.is_numeric_dtype(data):
    file_context = lib.virtualfile_from_matrix(data)
```

But if there **is** a text column, then the data input will have to be a `pandas.DataFrame`:

```python
else:
    file_context = lib.virtualfile_from_vectors(
        *[data[column] for column in data]
    )
```
Will need to work out how to loop through columns in both pandas and numpy arrays in the same manner. The key bit is `*[data[column] for column in data]` (which only works for `pandas`).

Reply:
On second thought, maybe it's not a good idea to allow `np.ndarray` input if a text column is given. Mixing numbers and text means the whole numpy array has an `np.object` data type which is memory-inefficient. The code below to handle either pandas or numpy arrays with text input is also not very elegant.

```python
                else:
                    _data = data if not hasattr(data, "iloc") else data.iloc
                    file_context = lib.virtualfile_from_vectors(
                        *[_data[:, column] for column in range(data.shape[1])]
                    )
```

Reply:
Hi @weiji14 , I rewrote this part. To exclude np.array with a text column. What do you think about this:
```python3
        with Session() as lib:
            # Choose how data will be passed in to the module
            if kind == "file":
                file_context = dummy_context(data)
            elif kind == "matrix":
                if pd.api.types.is_numeric_dtype(data):
                    file_context = lib.virtualfile_from_matrix(data)
                elif isinstance(data,np.ndarray):
                    raise GMTInvalidInput(
                        "Text columns are not supported with numpy array. "
                        "They are only supported with file or "
                        "pandas dataframe inputs."
                    )
                elif isinstance(data,pd.core.frame.DataFrame):
                    file_context = lib.virtualfile_from_vectors(
                        *[data[column] for column in data]
                    )
                else:
                    raise GMTInvalidInput("Unrecognized data type: {}".format(type(data)))
```